test/table.ts

```diff
@@ -1109,350 +1125,154 @@ describe('Bigtable/Table', () => {
-      // stream.end(); }, ...];
-      let makeRetryableError: Function;
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      let reqOptsCalls: any[];
-      let setTimeoutSpy: sinon.SinonSpy;
-
-      beforeEach(() => {
-        FakeChunkTransformer.prototype._transform = function (
-          rows: Row[],
-          enc: {},
-          next: Function
-        ) {
-          rows.forEach(row => this.push(row));
-          this.lastRowKey = rows[rows.length - 1].key;
-          next();
-        };
-
-        FakeChunkTransformer.prototype._flush = (cb: Function) => {
-          cb();
-        };
-
-        callCreateReadStream = (options: {}, verify: Function) => {
-          table.createReadStream(options).on('end', verify).resume(); // The stream starts paused unless it has a `.data()`
-          // callback.
-        };
-
-        emitters = null; // This needs to be assigned in each test case.
-
-        makeRetryableError = () => {
-          const error = new Error('retry me!') as ServiceError;
-          error.code = 4;
-          return error;
-        };
-
-        (sandbox.stub(FakeFilter, 'createRange') as sinon.SinonStub).callsFake(
-          (start, end) => {
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            const range: any = {};
-            if (start) {
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any
-              range.start = (start as any).value || start;
-              range.startInclusive =
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                typeof start === 'object' ? (start as any).inclusive : true;
-            }
-            if (end) {
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any
-              range.end = (end as any).value || end;
-            }
-            return range;
-          }
-        );
-
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (FakeMutation as any).convertToBytes = (value: string) => {
-          return Buffer.from(value);
-        };
-
-        reqOptsCalls = [];
-
-        setTimeoutSpy = sandbox
-          .stub(global, 'setTimeout')
-          .callsFake(fn => (fn as Function)());
-
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        table.bigtable.request = (config: any) => {
-          reqOptsCalls.push(config.reqOpts);
-
-          const stream = new PassThrough({
-            objectMode: true,
-          });
-
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          (stream as any).abort = () => {};
-
-          setImmediate(() => {
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            (emitters!.shift() as any)(stream);
-          });
-          return stream;
-        };
+    describe('createReadStream mocking out the gapic layer', () => {
```

Comment:
These are unit tests that ensure the right data reaches the Gapic layer.

Reply:
Request for a Bigtable SDK person - I'm not sure what the norm is for mocking out data in the gapic layer - I want to make sure that the tests are testing what they need to but not overstepping their bounds and with a partially handwritten library I don't really know where that line gets drawn

Reply:
The tests in this file will not be testing any of the generated code. The new tests just ensure that once the data reaches the generated code that the data is correct.